packages/react/src/file-input/FileInput.tsx

```diff
@@ -104,57 +116,56 @@ export const FileInput = forwardRef<HTMLInputElement, FileInputProps>(
 				id={id}
 			>
 				{(a11yProps) => (
-					<input
-						accept={accept?.toString()}
-						ref={ref}
-						css={styles}
-						{...a11yProps}
-						type="file"
-						disabled={disabled}
-						onChange={onChange}
-						{...props}
-					/>
+					<>
+						<Flex gap={1} alignItems="center">
+							<Button
+								{...a11yProps}
+								aria-invalid={undefined} // Buttons not announced as invalid when `aria-invalid`, add announcement to label instead
+								aria-label={`${label}. ${
+									a11yProps['aria-invalid'] ? 'Invalid. ' : ''
+								}${a11yProps['aria-required'] ? 'Required. ' : ''}Select file`}
+								aria-required={undefined} // Buttons can't be `aria-required`, add announcement to label instead
+								css={{ minWidth: 'max-content' }}
+								disabled={disabled}
+								onClick={onVisualButtonClick}
+								variant="secondary"
+							>
+								Select file
+							</Button>
+
+							<div role="status">
```

Comment:
I think you'll also have to add this to the label or describe. I didn't get an announcement in my VO btw

Reply:
Hmm describe is a good idea, thanks.

- [ ] aria-describedby rather than status